frontend/catalyst/compilation_pipelines.py

```diff
@@ -688,13 +768,30 @@ def get_cmain(self, *args):
         return function.get_cmain(*args)
 
     def __call__(self, *args, **kwargs):
+        static_argnums = self.compile_options.static_argnums
+
         if EvaluationContext.is_tracing():
             return self.user_function(*args, **kwargs)
 
         function, args = self._ensure_real_arguments_and_formal_parameters_are_compatible(
             self.compiled_function, *args
         )
-        recompilation_needed = function != self.compiled_function
+
+        recompilation_needed = (
```

Comment:
Minor suggestion, but this variable always confuses me because it makes it sound like recompilation still needs to happen, when in fact it already happened in `_ensure_real_arguments_and_formal_parameters_are_compatible` (not a great name either I know 😏).

Maybe `recompilation_happened` is clearer?

Reply:
Agree! The variable is a bit confusing since the recompilation should happen before it.